### PR TITLE
HasSortableRelations trait

### PR DESCRIPTION
**Pull request by mjauvin**
- Add missing relation events. (replaces #20)
- Port @tobias-kuendig PR to add SortableRelation trait

ref. https://github.com/octobercms/library/pull/526

Fixes https://github.com/wintercms/winter/issues/33

### CODE REVIEW

**Comment by LukeTowers** [comment]
Is there not some way of identifying the sortable information from the related model's own usage of sortable traits?

**Comment by mjauvin** [comment]
@bennothommo what are those failing tests?

File: src/Database/Traits/SortableRelation.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ *   use \Winter\Storm\Database\Traits\SortableRelation;
+ *
+ *   public $sortableRelations = ['relation_name' => 'sort_order_column'];
+ *
+ * To set orders:
+ *
+ *   $model->setSortableRelationOrder($relationName, $recordIds, $recordOrders);
+ *
+ */
+trait SortableRelation
```
Comment (LukeTowers) [inline src/Database/Traits/SortableRelation.php]: This should probably be `HasSortableRelations`

Comment (mjauvin) [reply src/Database/Traits/SortableRelation.php]: done

**Comment by mjauvin** [comment]
@LukeTowers I've changed the code path you didn't like and added an event instead to handle this gracefully for non-pivot based relations.

**Comment by bennothommo** [comment]
Replaced by #94 on an internal branch.

**Pull request closed without merging**